src/waveform/vsyncthread.cpp

```diff
@@ -46,16 +53,72 @@ void VSyncThread::run() {
             m_sinceLastSwap = m_timer.restart();
             m_waitToSwapMicros = 1000;
             usleep(1000);
-        } else { // if (m_vSyncMode == ST_TIMER) {
+        } else if (m_vSyncMode == ST_PLL) {
+            // Use a phase-locked-loop on the QOpenGLWindow::frameSwapped signal
+            // to determine when the vsync occurs
+
+            // last estimated vsync
+            auto pllPhaseOut = std::llround(m_pllPhaseOut);
+            // estimated frame interval
+            auto pllDeltaOut = std::llround(m_pllDeltaOut);
+
+            auto now = m_frameSwappedTimer.elapsed().toIntegerMicros();
+            auto sincePllPhaseOut = now - pllPhaseOut;
+
+            // qDebug() << "PLL sincePllPhaseOut:" << sincePllPhaseOut;
+
+            if (sincePllPhaseOut > pllDeltaOut) {
+                // if the timer had advanced more than a frame since the last
+                // estimated vsync, move the estimation to just before now
+                // (this happens when the QOpenGLWindow updateRequest and
+                // frameSwapped signals are not delivered in time)
+                pllPhaseOut += (sincePllPhaseOut / pllDeltaOut) * pllDeltaOut;
+                sincePllPhaseOut = now - pllPhaseOut;
+            }
+
+            // TODO we could use the pll related timer and variables directly
+            // for the visual playback position, but for now use the existing
+            // mechanism:
+            m_sinceLastSwap = m_timer.restart();
+            m_waitToSwapMicros = pllDeltaOut - sincePllPhaseOut;
+            m_syncIntervalTimeMicros = pllDeltaOut;
+
+            // qDebug() << "PLL m_waitToSwapMicros" << m_waitToSwapMicros;
+
+            int signalled = 0;
+            // Signal to render and swap the gl widgets (waveforms, spinnies, vumeters)
+            // This works best when offset with a delay wrt the vsync time.
+            const int offset = 8000; // Found by trial and error
```

Comment:
I set this to 12000 and it looks much better on Windows (with setSwapInterval=1). I would say better than before, but not perfect.

Reply:
Interesting, so with setSwapInterval(0) you get 160 fps from the phase locked loop.

When you say better than before, do you mean better than with this PR with setSwapInterval(0), or better than without this PR ?

Reply:
I meant, this PR + setSwapInterval(1) + offset=12000 was slightly better than 2.4. All other combinations were worse.
I haven't tried your last commit.